core/src/main/java/org/fao/geonet/resources/CMISResources.java

```diff
@@ -0,0 +1,491 @@
+    }
+
+    @Override
+    Pair<byte[], Long> loadResource(final Path resourcesDir, final ServletContext context,
+                                    final Path appPath, final String filename, final byte[] defaultValue,
+                                    final long loadSince) throws IOException {
+        final Path file = locateResource(resourcesDir, context, appPath, filename);
+        final String key = getKey(file);
+        try {
+            final CmisObject object = CMISConfiguration.getClient().getObjectByPath(key);
+            if (object != null) {
+                final long lastModified = object.getLastModificationDate().toInstant().toEpochMilli();
+                try (InputStream in = ((Document) object).getContentStream().getStream()) {
+                    if (loadSince < 0 || lastModified > loadSince) {
+                        byte[] content = new byte[(int) ((Document) object).getContentStreamLength()];
+                        new DataInputStream(in).readFully(content);
+                        return Pair.read(content, lastModified);
+                    } else {
+                        return Pair.read(defaultValue, loadSince);
+                    }
+                }
+            } else {
+                Log.info(Log.RESOURCES, "Error loading resource " + CMISConfiguration.getRepositoryId() + ":" + key);
+            }
+        } catch (CmisObjectNotFoundException e) {
+            Log.warning(Geonet.RESOURCES,
+                    String.format("Unable to locate resource '%s'.", key));
+            // Ignore not found error.
+        }
+        return Pair.read(defaultValue, -1L);
+    }
+
+    @Override
+    protected Path locateResource(@Nullable final Path resourcesDir, final ServletContext context,
+                                  final Path appPath, @Nonnull String filename) throws IOException {
+        if (filename.charAt(0) == '/' || filename.charAt(0) == '\\') {
+            filename = filename.substring(1);
+        }
+
+        final String key;
+        if (resourcesDir != null) {
+            key = getKey(resourcesDir, filename);
+        } else {
+            key = CMISConfiguration.getFolderDelimiter() +  filename;
+        }
+
+        boolean keyExists=false;
+        // Use getObjectByPath as it does caching while existsPath does not.
+        try {
+            CMISConfiguration.getClient().getObjectByPath(key);
+            keyExists = true;
+        } catch (CmisObjectNotFoundException e) {
+            keyExists=false;
+        }
+
+        if (!keyExists) {
+            Path webappCopy = null;
+            if (context != null) {
+                final String realPath = context.getRealPath(filename);
+                if (realPath != null) {
+                    webappCopy = IO.toPath(realPath);
+                }
+            }
+
+            if (webappCopy == null) {
+                webappCopy = appPath.resolve(filename);  // TODO: this won't work...
```

Comment:
Is this something to solve?

Reply:
Copied from S3.  Code was updated in latest changes.